pandas/tests/groupby/test_indexing.py

```diff
@@ -314,3 +314,18 @@ def test_groupby_duplicated_columns(func):
         [[1.5, 3.0, 1.5]], columns=["A", "B", "A"], index=pd.Index(["G"], name="C")
     )
     tm.assert_frame_equal(result, expected)
+
+
+def test_dataframe_groupby_incorrect_with_multiindex_make_df_from_data():
```

Comment:
Could you give a better name? Something like test_groupby_get_nonexisting_groups